src/layout/umap.c

```diff
@@ -161,26 +274,26 @@ static igraph_error_t igraph_i_umap_find_prob_graph(const igraph_t *graph,
          * to avoid double counting.
          *
          * To implement that, we keep track of whether we have "seen" this edge before.
-         * If no, set the P_{->}. If yes, it contains P_{->} and now we can substitute
+         * If not, set the P_{->}. If yes, it contains P_{->} and now we can substitute
          * it with the final result.
          *
          * */
-        for (igraph_integer_t j = 1; j < no_of_neis; j++) {
+        for (igraph_integer_t j = 0; j < no_of_neis; j++) {
             eid = VECTOR(eids)[j];
             /* Basically, nodes closer than rho have probability 1, but nothing disappears */
             weight = sigma < 0 ? 1: exp(-(VECTOR(*distances)[eid] - rho) / sigma);
```

Comment:
```suggestion
            weight = sigma < 0 ? 1 : exp(-(VECTOR(*distances)[eid] - rho) / sigma);
```

(cosmetic nitpick, sorry)